### PR TITLE
Add Lynx

### PR DESCRIPTION
**Pull request by JcMinarro**
Add Lynx to project. Lynx is a library that show a customView with the logcat of your application, and allows you to search and filter on it.

### CODE REVIEW

File: app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java [hunk 1]
```diff
@@ -197,6 +200,11 @@ public void showAppNeedsToBeRestarted() {
         runOnUiThreadIfFragmentAlive(() -> Toast.makeText(getContext(), "To apply new settings app needs to be restarted", Toast.LENGTH_LONG).show());
     }
 
+    @OnClick(R.id.tv_show_log)
+    void showLog(View view) {
+        view.getContext().startActivity(LynxActivity.getIntent(view.getContext(), new LynxConfig()));
```
Comment (vanniktech) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: view.getContext() could be extracted

Comment (vanniktech) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Also should not we provide the LynxConfig via Dagger?

Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Nice library! I'll take a look tonight. Please create an issue to discuss
things before PR next time :)

On Tue, Dec 29, 2015, 13:56 Niklas Baudy <notifications@github.com> wrote:

> In
> app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java
> <https://github.com/artem-zinnatullin/qualitymatters/pull/77#discussion_r48533530>
> :
>
> > @@ -197,6 +200,11 @@ public void showAppNeedsToBeRestarted() {
> >          runOnUiThreadIfFragmentAlive(() -> Toast.makeText(getContext(), "To apply new settings app needs to be restarted", Toast.LENGTH_LONG).show());
> >      }
> >
> > +    @OnClick(R.id.tv_show_log)
> > +    void showLog(View view) {
> > +        view.getContext().startActivity(LynxActivity.getIntent(view.getContext(), new LynxConfig()));
>
> Also should not we provide the LynxConfig via Dagger?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artem-zinnatullin/qualitymatters/pull/77/files#r48533530>
> .
>
-- 

@artem_zin

Comment (JcMinarro) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Ok, I will create a new Issue next time ;)

**Comment by vanniktech** [comment]
:+1: from my side. If @artem-zinnatullin also agrees you shoul squash your commits into one

**Comment by artem-zinnatullin** [comment]
Yup, please squash!

Sorry for delay with review from my side, New Year stuff.. Will review
asap!

On Wed, Dec 30, 2015, 23:37 Niklas Baudy <notifications@github.com> wrote:

> [image: :+1:] from my side. If @artem-zinnatullin
> <https://github.com/artem-zinnatullin> also agrees you shoul squash your
> commits into one
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artem-zinnatullin/qualitymatters/pull/77#issuecomment-168069875>
> .
>
-- 

@artem_zin

File: README.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ Also, you might notice that app has custom [`functional test runner`](/app/src/f
 * [Stetho](http://facebook.github.io/stetho/) — inspect the app via Chromium Developer Tools (network requests, db, preferences and so on). Must have for developers.
 * [LeakCanary](https://github.com/square/leakcanary) — detect memory leaks without IDE! Must have for QAs and developers.
 * [TinyDancer](https://github.com/brianPlummer/TinyDancer) — see frame rate right on your screen. Must have for QAs and developers.
+* [Lynx](https://github.com/pedrovgs/Lynx) — show a custom view with all the information logcat.
```
Comment (artem-zinnatullin) [inline README.md]: Maybe `show a custom view with LogCat output`?

Comment (artem-zinnatullin) [inline README.md]: Or even `see LogCat output right in the app, useful for QAs and developers.`

File: app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java [hunk 3]
```diff
@@ -197,6 +204,12 @@ public void showAppNeedsToBeRestarted() {
         runOnUiThreadIfFragmentAlive(() -> Toast.makeText(getContext(), "To apply new settings app needs to be restarted", Toast.LENGTH_LONG).show());
     }
 
+    @OnClick(R.id.tv_show_log)
+    void showLog(View view) {
```
Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: `@NonNull`

Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Wait, why not just pick activity from fragment?

File: app/src/debug/res/layout/fragment_developer_settings.xml [hunk 4]
```diff
@@ -157,6 +157,14 @@
 
         </LinearLayout>
 
+        <Button
```
Comment (artem-zinnatullin) [inline app/src/debug/res/layout/fragment_developer_settings.xml]: Can you please use same pattern with `LinearLayout`, styles and `TextView` on the left as other developer tools? For the button itself you can use style `?android:attr/borderlessButtonStyle` and write `show` as text, in `TextView` on the left you can write something like `LogCat`

File: app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java (hunk 3 above)
Comment (vanniktech) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: If he does there's no need for the view anymore

File: app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java [hunk 5]
```diff
@@ -68,4 +69,10 @@ public DeveloperSettingsPresenter provideDeveloperSettingsPresenter(@NonNull Dev
                                                                         @NonNull AnalyticsModel analyticsModel) {
         return new DeveloperSettingsPresenter(developerSettingsModelImpl, analyticsModel);
     }
+
+    @NonNull
+    @Provides
```
Comment (vanniktech) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: Maybe add @Singleton so we'd reuse the Config object. @artem-zinnatullin what do you think about that?

Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: I'm okay with `@Singleton` but IDK if `LynxConfig` is immutable or not

Comment (JcMinarro) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: LynxConfig is mutable

Comment (JcMinarro) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: What happen with it at the end??

Comment (JcMinarro) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: @vanniktech @artem-zinnatullin  What happen with it at the end??

Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java]: Jeez, sorry, reviewing :)

On Tue, 2 Feb 2016, 13:43 Jc Miñarro <notifications@github.com> wrote:

> In
> app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModule.java
> <https://github.com/artem-zinnatullin/qualitymatters/pull/77#discussion_r51551763>
> :
>
> > @@ -68,4 +69,10 @@ public DeveloperSettingsPresenter provideDeveloperSettingsPresenter(@NonNull Dev
> >                                                                          @NonNull AnalyticsModel analyticsModel) {
> >          return new DeveloperSettingsPresenter(developerSettingsModelImpl, analyticsModel);
> >      }
> > +
> > +    @NonNull
> > +    @Provides
>
> @vanniktech <https://github.com/vanniktech> @artem-zinnatullin
> <https://github.com/artem-zinnatullin> What happen with it at the end??
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artem-zinnatullin/qualitymatters/pull/77/files#r51551763>
> .
>
-- 

@artem_zin

**Comment by artem-zinnatullin** [comment]
:+1:

**Comment by artem-zinnatullin** [comment]
Thanks!

**Comment by vanniktech** [comment]
Yup thanks @JcMinarro

**Comment by JcMinarro** [comment]
:+1:

**Pull request merged**